### PR TITLE
Add the ability for users to support different ignore_paths settings for different requests

### PR DESCRIPTION
**Pull request by hythof**
### Purpose/goal of Pull Request (Issue #)
Relate: https://github.com/WOVNio/wovnrb/pull/94

The user can modify the Rack::Request object to change the token on a per-request basis.
This is example for sinatra. Also other rack application can do it by simuler way.
```
require 'sinatra/base'

class Server < Sinatra::Base
  get '/dynamic_ignore' do
    request.update_param('wovn_ignore_paths', ['/dynamic_ignore', '/other_rule'])
    erb :index
  end 
end
```

### Comments
`wovn_token` and this one can be used together.

### CODE REVIEW

File: lib/wovnrb/store.rb [hunk 1]
```diff
@@ -185,14 +178,47 @@ def wovn_host
...
-    end
 
     def stringify_keys!(h)
       h.keys.each do |k|
         h[k.to_s] = h.delete(k)
       end
     end
   end
+
+  class Settings < Hash
```
Comment (mrasu) [inline lib/wovnrb/store.rb]: why not divide to different file?

File: lib/wovnrb/store.rb [hunk 2]
```diff
@@ -185,14 +178,47 @@ def wovn_host
...
+    def initialize(*args, **kwargs)
+      super(*args, **kwargs)
+      @dynamic = {}
```
Comment (mrasu) [inline lib/wovnrb/store.rb]: `dynamic` is adjective.
why not change name to dynamic_setting or something noun?

File: lib/wovnrb/store.rb [hunk 3]
```diff
@@ -185,14 +178,47 @@ def wovn_host
...
+
+    def update_dynamic!(params)
+      # If the user defines dynamic settings for this request, use it instead of the config
+      DYNAMIC.each do |params_key, setting_key|
+        value = params[params_key]
+        @dynamic[setting_key] = value if value
+      end
+    end
+
+    DYNAMIC = {
```
Comment (mrasu) [inline lib/wovnrb/store.rb]: please not use the same name in same scope (`@dynamic` and `DYNAMIC`)

File: lib/wovnrb/store.rb (hunk 1 above)
Comment (hythof) [reply lib/wovnrb/store.rb]: That's right.
I made `settings.rb`.

File: lib/wovnrb/store.rb (hunk 2 above)
Comment (hythof) [reply lib/wovnrb/store.rb]: I renamed to `dynamic_settings`

File: lib/wovnrb/store.rb (hunk 3 above)
Comment (hythof) [reply lib/wovnrb/store.rb]: It's renamed `DYNAMIC_KEYS`.
Thanks.

File: lib/wovnrb.rb [hunk 4]
```diff
@@ -50,11 +51,9 @@ def call(env)
 
         request = Rack::Request.new(env)
         unless request.params['wovn_disable'] == true
+          @store.settings.update_dynamic!(request.params)
```
Comment (a-bougouin) [inline lib/wovnrb.rb]: Could you rename into `@store.settings.update_dynamic_settings!` to be consistant with `@store.settings.clear_dynamic_settings!` please?

Comment (hythof) [reply lib/wovnrb.rb]: I renamed it.

**Pull request merged**